game/plugins/src/main/kotlin/gg/rsmod/plugins/content/mechanics/trading/trade_script.kts

```diff
@@ -0,0 +1,179 @@
+        partner.getTradeRequests().add(player)
+
+        // Send the trade request
+        player.message("Sending trade request...")
+        partner.message(TRADE_REQ_STRING.format(player.username), ChatMessageType.TRADE_REQ)
+    } else {
+
+        // Remove the requests
+        player.getTradeRequests().remove(partner)
+        partner.getTradeRequests().remove(player)
+
+        // Initiate the trade
+        initiate(player, partner)
+    }
+}
+
+/**
+ * Initiates a trade between two players
+ *
+ * @param player    The first player
+ * @param partner   The partner player
+ */
+fun initiate(player: Player, partner: Player) {
+
+    // The trade session instances
+    val playerSession = TradeSession(player, partner)
+    val partnerSession = TradeSession(partner, player)
+
+    // Define the session attribute for both players
+    player.attr[TRADE_SESSION_ATTR] = playerSession
+    partner.attr[TRADE_SESSION_ATTR] = partnerSession
+
+    // Initialise the sessions
+    playerSession.open()
+    partnerSession.open()
+}
+
+// Item Offer Event
+on_button(OVERLAY_INTERFACE, 0) {
+    val trade = player.getTradeSession()
+    trade?.let {
+
+        // The player's inventory
+        val inventory = player.inventory
+
+        // The item slot, and the option that was pressed
+        val slot = player.getInteractingSlot()
+        val opt = player.getInteractingOption()
+
+        // The item being traded
+        val item = inventory[slot] ?: return@on_button
+
+        // Queue the action, as we might need to access queued dialogue
+        player.queue(TaskPriority.WEAK) {
+
+            // The amount being traded
+            val amount = when (opt) {
+                2 -> 5
+                3 -> 10
+                4 -> inventory.getItemCount(item.id)
+                5 -> inputInt("Enter amount:")
+                else -> 1
+            }
+
+            // Offer the amount to the trade
+            player.getTradeSession()?.offer(slot, amount)
+        }
+    }
+}
+
+// Item Remove Event
+on_button(TRADE_INTERFACE, PLAYER_TRADE_CHILD) {
+    val trade = player.getTradeSession()
+    trade?.let {
+
+        // The player's trade container
+        val container = it.container
+
+        // The item slot, and the option that was pressed
+        val slot = player.getInteractingSlot()
+        val opt = player.getInteractingOption()
+
+        // The item being traded
+        val item = container[slot] ?: return@on_button
+
+        // Queue the action, as we might need to access queued dialogue
+        player.queue(TaskPriority.WEAK) {
+
+            // The amount being traded
+            val amount = when (opt) {
+                2 -> 5
+                3 -> 10
+                4 -> container.getItemCount(item.id)
+                5 -> inputInt("Enter amount:")
+                else -> 1
+            }
+
+            // Offer the amount to the trade
+            player.getTradeSession()?.remove(slot, amount)
```

Comment:
You're already null-checking `getTradeSession` on line 123.
`it.remove(slot, amount)`

Reply:
You're right - it was late and I didn't pick up on this - `it` in this context is actually the `CoroutineContext` - I'll just explicitly declare the parameter as `trade` for the scope of the null-check